### PR TITLE
Added timestamps to packet viewer

### PR DESCRIPTION
**Pull request by James-Money**
- Added timestamps to see when certain packets happen

### CODE REVIEW

File: src/parser.js [hunk 1]
```diff
@@ -208,6 +208,10 @@ class NEXParser extends EventEmitter {
 
 				connection.checkForSecureServer = false;
 			}
+			let date;
+			const timestamp = raw.header.timestampSeconds;
+			date = new Date(timestamp * 1000);
+			packet.date = date;
```
Comment (jonbarrow) [inline src/parser.js:214]:
```suggestion
			const timestamp = raw.header.timestampSeconds;
			packet.date = new Date(timestamp * 1000);
```

Also this only works for packets coming from the `pcap-parser` library. `pcap-ng-parser` encodes timestamp data differently https://www.npmjs.com/package/pcap-ng-parser#event-data

File: src/packet.js [hunk 2]
```diff
@@ -77,7 +78,8 @@ class Packet {
 				errorCode: this.rmcMessage.errorCode,
 				body: this.rmcData.body
 			},
-			stackTrace: this.stackTrace
+			stackTrace: this.stackTrace,
+			date: this.date
```
Comment (jonbarrow) [inline src/packet.js:82]: `this.date` should also be initialized in the constructor

**Review by jonbarrow — CHANGES_REQUESTED** [review]
Front end looks fine. But the backend needs some changes. Most importantly, the current implementation only works for for the `pcap-parser` library and will break on `pcapng` dumps

While `pcapng` dumps are less common now, since Hokaku creates `pcap` files, for historic reasons it's important to keep this functionality. Also there may be cases where Hokaku is not used, such as in some 3DS titles where Hokaku is not supported

File: src/packet.js (hunk 2 above)
Comment (James-Money) [reply src/packet.js:82]: If I were to replace line 52 in the constructor with `this.date = date;`, it causes the list to no longer populate when opening a file. If I were to use `date = this.date;`, the variable goes unused. Should I replace it with `this.date = 0;`?

Comment (jonbarrow) [reply src/packet.js:82]: Yes `this.date = 0;` should be fine

**Comment by James-Money** [comment]
I've implemented proper support for grabbing timestamps from `.pcapng` files. It should be noted that captures taken from Wireshark are recorded in UTC, and will display as such in the tool.

**Pull request merged**